test/functional/test_framework/key.py

```diff
@@ -431,6 +471,41 @@ def sign_schnorr(self, msg):
         e = int.from_bytes(hashlib.sha256(R[0].to_bytes(32, 'big') + self.get_pubkey().get_bytes() + msg).digest(), 'big') % SECP256K1_ORDER
         return R[0].to_bytes(32, 'big') + ((k + e*self.secret) % SECP256K1_ORDER).to_bytes(32, 'big')
 
+    def sign_schnorr_with_nonce(self, msg, k_key):
```

Comment:
This is almost identical to the code above in `sign_schnorr()`. Can you just add an optional parameter `nonce=None` to `sign_schnorr()` instead of copy-pasting a lot of code?